flatland/core/effects_generator.py

```diff
@@ -0,0 +1,31 @@
+from typing import Callable, TypeVar
+
+from flatland.core.env import Environment
+
+EnvironmentType = TypeVar('EnvironmentType', bound=Environment, covariant=True)
+
+
+class EffectsGenerator[U]:
+    """
+    PostHook for external events modifying the env (state) before observations and rewards are computed.
+
+    See https://github.com/flatland-association/flatland-workshop-2024/blob/main/next-flatland/documentation/core_concept.md#effect-creator
```

Comment:
@manuschn should it not rather be a pre-hook as in the GenEnv proposal? This would be in line with malfunction generator triggered at beginning of env step.